cartographer/mapping/internal/2d/pose_graph_2d.cc

```diff
@@ -1240,12 +1240,31 @@ void PoseGraph2D::TrimmingHandle::TrimSubmap(const SubmapId& submap_id) {
   // Remove all 'data_.constraints' related to 'nodes_to_remove'.
   {
     std::vector<Constraint> constraints;
+    std::set<SubmapId> unconstrained_submap_ids;
     for (const Constraint& constraint : parent_->data_.constraints) {
       if (nodes_to_remove.count(constraint.node_id) == 0) {
         constraints.push_back(constraint);
+      } else {
+        // A constraint to another submap will be removed, mark it as affected.
+        unconstrained_submap_ids.insert(constraint.submap_id);
       }
     }
     parent_->data_.constraints = std::move(constraints);
+    // Go through the changed constraints and finalize the list of submap IDs
+    // that lost constraints and now have zero inter-submap constraints.
```

Comment:
And here you reduce this list to the submaps which lost all of them. Can we clarify the comment?

Reply:
I simplified all comments, I hope it's clearer now.